.github/workflows/integration.yaml

```diff
@@ -0,0 +1,105 @@
+name: 'Integration'
+
+on:
+  push:
+    branches:
+    - 'main'
+  pull_request:
+    branches:
+    - 'main'
+  workflow_dispatch:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+permissions:
+  contents: 'read'
+  id-token: 'write'
+
+env:
+  REGION: us-central1
+
+jobs:
+  test:
+    if: ${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name && github.actor != 'dependabot[bot]' }}
+    runs-on: 'ubuntu-latest'
+
+    steps:
+    - id: 'checkout'
+      name: 'Checkout'
+      uses: 'actions/checkout@v3'
+
+    - id: 'compute-test-uid'
+      name: 'Compute test UID'
+      run: |-
+        echo "TEST_UID=${GITHUB_JOB}-${GITHUB_SHA::7}-${GITHUB_RUN_NUMBER}" >> ${GITHUB_ENV}
+
+    - id: 'set-up-node'
+      name: 'Set up Node'
+      uses: 'actions/setup-node@v3'
+      with:
+        node-version: '16.x'
+
+    - id: 'build'
+      name: 'Build'
+      run: 'npm ci && npm run build'
+
+    - id: 'auth'
+      name: 'Auth'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        workload_identity_provider: '${{ secrets.WIF_PROVIDER_NAME }}'
+        service_account: '${{ secrets.SERVICE_ACCOUNT_EMAIL }}'
+
+    - id: 'gcloud-setup'
+      name: 'Set up Cloud SDK'
+      uses: google-github-actions/setup-gcloud@v0
```

Comment:
For our actions, we pin to `@main` (yea, it'll generate a warning, but that's okay)